### DISCUSSION TITLE
Zustand with SWR

### DISCUSSION THREAD

**Discussion by mawnir** [pmndrs/zustand#2280] [Q&A]
i fetch data with: `const { data, error, isLoading } = useSWR("/api/documents/${params.documentId}", fetcher );`
i need to put  `data` in `document`: `const { document, setDocument } = useDocumentStore();` ==>  a Zustand store
when i do:
```
 if (data) {
    setDocument(data);
  } 
```

> i get: **infinite loop**

and when i do:
 ```
 useEffect(() => {
     if (data && !isLoading) {
       setDocument(data);
     }
   }, [setDocument, data, isLoading]);
```

> i get : **React has detected a change in the order of Hooks..**

how can i fix this?

**Comment by dbritto-dev** [comment]
@mawnir would you mind sharing us a minimal repro on codesandbox?

**Comment by mawnir** [comment]
> @mawnir would you mind sharing us a minimal repro on codesandbox?

after creating the sandbox, i found my error, it was the position of useEffect.. thank you!   
https://codesandbox.io/p/devbox/zustand-swr-tristan-q8ghng?file=%2Fsrc%2FApp.tsx%3A8%2C1-9%2C1
